lib/Ix/DBIC/ResultSet.pm

```diff
@@ -471,6 +449,29 @@ sub _ix_check_user_properties (
       }
     }
 
+    if ($date_fields{$prop}) {
+      unless (defined $value) {
+        if ($prop_info->{$prop}->{is_optional}) {
+          $user_prop{$prop} = $value;
+          next PROP;
+        } else {
+          $property_error{$prop} = "no value given for required field";
+          next PROP;
+        }
+      }
+
+      # Already a DateTime object (checked above)?
+      unless (ref $value) {
+        if (my $dt = parsedate($value)) {
+          # great, it's already valid
+          $value = $dt;
+        } else {
+          $property_error{$prop} = "invalid date value";
+          next PROP;
+        }
+      }
+    }
+
```

Comment:
I think there's a bigger but related issue.  With your change, there are two places where we can skip the a field for being undefined: once at line 459 if it's an undefined where we expect a mandatory date, and once at 487 when it's the generic check used for undefined mandatory properties.

This seems like needless duplication.  We only need to check the definedness once, when we check it for everything else.

Then I wonder whether the datetime check shouldn't go before the validator.  Since it normalizes input strings into objects, then validator checks (like `$info->{validator}` stores) could be written expecting DateTime objects, so you could write a validator to only allow dates in the future.

Does that make sense?